source/globalCommands.py

```diff
@@ -2008,18 +2023,20 @@ def script_review_currentSymbol(self,gesture):
 	@script(
 		description=_(
 			# Translators: Input help mode message for toggle speech mode command.
-			"Toggles between the speech modes of off, beep and talk. "
+			"Toggles between the speech modes of off, beep, talk and on-demand. "
 			"When set to off NVDA will not speak anything. "
 			"If beeps then NVDA will simply beep each time it its supposed to speak something. "
 			"If talk then NVDA will just speak normally."
+			"If on-demand then NVDA will only speak when calling commands whose goal is to speak something, but not"

```

Comment:
I think this may be too long for comfort in input help. Perhaps:

```suggestion
			"If on-demand, NVDA will only speak when commands which have no purpose other than to provide feedback are used. "
```